src/m365/spo/commands/listitem/listitem-roleassignment-add.spec.ts

```diff
@@ -0,0 +1,421 @@
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if upn and groupName are specified', async () => {
+    const actual = await command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listTitle: 'Documents', listItemId: 1, upn: 'someaccount@tenant.onmicrosoft.com', groupName: 'someGroup', roleDefinitionId: 1073741827 } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if roleDefinitionId and roleDefinitionName are specified', async () => {
+    const actual = await command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listTitle: 'Documents', listItemId: 1, groupName: 'someGroup', roleDefinitionId: 1073741827, roleDefinitionName: 'readers' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation neither roleDefinitionId nor roleDefinitionName is specified', async () => {
+    const actual = await command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listTitle: 'Documents', listItemId: 1, groupName: 'someGroup'} }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation neither groupName nor principalId or upn is specified', async () => {
+    const actual = await command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listTitle: 'Documents', listItemId: 1, roleDefinitionName: 'readers' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation neither listTitle nor listId or listUrl is specified', async () => {
+    const actual = await command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', groupName: 'someGroup', listItemId: 1, roleDefinitionName: 'readers' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the listItemId option is not a number', async () => {
+    const actual = await command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listTitle: 'Documents', groupName: 'someGroup', listItemId: 'abc', roleDefinitionName: 'readers' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the listItemId option is a number', async () => {
+    const actual = await command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listTitle: 'Documents', groupName: 'someGroup', listItemId: 1, roleDefinitionName: 'readers' } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('add role assignment to listitem in list by title and role definition id', async () => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if ((opts.url as string).indexOf('_api/web/lists/getByTitle(\'test\')/items(1)/roleassignments/addroleassignment(principalid=\'11\',roledefid=\'1073741827\')') > -1) {
```

Comment:
Let's try to avoid the usage of `indexOf` and in place use a full comparison.
`if (opts.url === 'exampleurl')`